### PR TITLE
Upgrade to Finagle 18.9.1

### PR DESCRIPTION
**Pull request by adleong**
This also upgrades Jackson to 2.9.6.

Most of the required changes were due to:
* Some exceptions moving from Failure to FailureFlags
* Writer is now parameterized instead of being fixed on Buf
* Reader.writable is replaced by Pipe
* Stack.Leaf and Stack.Node constructors are no longer accessible and replaced by methods Stack.leaf and Stack.node
* Counters are now eagerly created with a value of 0 instead of only being created when first incremented
* Jackson's json mapping exception removed

### CODE REVIEW

File: namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ThriftInterpreterInterfaceConfig.scala [hunk 1]
```diff
@@ -51,7 +50,7 @@ class ThriftInterpreterInterfaceInitializer extends InterfaceInitializer {
   val configClass = classOf[ThriftInterpreterInterfaceConfig]
 }
 
-case class ThriftServable(addr: InetSocketAddress, iface: ThriftService, params: Stack.Params) extends Servable {
+case class ThriftServable(addr: InetSocketAddress, iface: AnyRef, params: Stack.Params) extends Servable {
```
Comment (dadjeibaah) [inline namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ThriftInterpreterInterfaceConfig.scala]: Could we still keep the strict type for `iface` as ThriftService using `import com.twitter.finagle.thrift.ThriftService`. Is there a reason why this should be `AnyRef`.

File: config/src/main/scala/io/buoyant/config/types/DirectoryDeserializer.scala [hunk 2]
```diff
@@ -10,10 +10,10 @@ case class Directory(path: java.nio.file.Path) {
 }
 
 class DirectoryDeserializer extends ConfigDeserializer[Directory] {
-  override def deserialize(jp: JsonParser, ctxt: DeserializationContext): Directory =
-    catchMappingException(ctxt) {
-      Directory(Paths.get(_parseString(jp, ctxt)))
-    }
+  override def deserialize(jp: JsonParser, ctxt: DeserializationContext): Directory = {
+    val dir = Directory(Paths.get(_parseString(jp, ctxt)))
```
[the remark this answers is not in the document]
Comment (adleong) [reply config/src/main/scala/io/buoyant/config/types/DirectoryDeserializer.scala]: the `_parseString` method no longer exists :(

File: finagle/h2/src/main/scala/com/twitter/finagle/buoyant/H2.scala [hunk 3]
```diff
@@ -26,7 +26,7 @@ object H2 extends Client[Request, Response] with Server[Request, Response] {
       stk.push(H2ApplicationProtocol.module)
       stk.push(TracingFilter.module)
       (StackClient.newStack[Request, Response] ++ stk.result)
-        .replace(StackClient.Role.pool, SingletonPool.module[Request, Response])
+        .replace(StackClient.Role.pool, SingletonPool.module[Request, Response](allowInterrupts = true))
```
[the remark this answers is not in the document]
Comment (adleong) [reply finagle/h2/src/main/scala/com/twitter/finagle/buoyant/H2.scala]: It has to do with propagating future interrupts but, TBH, I copied this setting value from Finagle's ThriftMux's singleton pool.

File: namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ThriftInterpreterInterfaceConfig.scala (hunk 1 above)
Comment (adleong) [reply namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ThriftInterpreterInterfaceConfig.scala]: com.twitter.finagle.thrift.ThriftService no longer exists

**Comment by adleong** [comment]
@dadjeibaah I've added a workaround for the import conflict and filed https://github.com/twitter/util/issues/229

I also had to disable the example tests because the CI job kept running out of memory.  We should investigate a long term solution for this so that we can re-enable those tests in CI.

File: config/src/main/scala/io/buoyant/config/types/DirectoryDeserializer.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (adleong) [reply config/src/main/scala/io/buoyant/config/types/DirectoryDeserializer.scala]: wait... I use `_parseString` in other deserializers.... what's going on.....

[the remark this answers is not in the document]
Comment (adleong) [reply config/src/main/scala/io/buoyant/config/types/DirectoryDeserializer.scala]: I got myself very confused, please ignore these comments.

I will fix the formatting here to not create a val.

**Pull request merged**